src/cargo/ops/cargo_compile.rs

```diff
@@ -284,12 +288,76 @@ pub fn compile_ws<'a>(
         unit_graph::emit_serialized_unit_graph(&bcx.roots, &bcx.unit_graph)?;
         return Ok(Compilation::new(&bcx)?);
     }
-
     let _p = profile::start("compiling");
     let cx = Context::new(&bcx)?;
     cx.compile(exec)
 }
 
+pub fn print<'a>(
+    ws: &Workspace<'a>,
+    options: &CompileOptions,
+    print_opt_value: &str,
+) -> CargoResult<()> {
+    let CompileOptions {
+        ref build_config,
+        ref target_rustc_args,
+        ..
+    } = *options;
+    let config = ws.config();
+    let rustc = config.load_global_rustc(Some(ws))?;
+    let mut kinds = build_config.requested_kinds.clone();
+    if kinds.is_empty() {
+        kinds.push(CompileKind::Host);
+    }
```

Comment:
This seems like it's likely duplicated with somewhere else (handling an empty set of kinds). If so can it be deduplicated?